src/server/game/Entities/Object/Object.cpp

```diff
@@ -2672,6 +2745,69 @@ float WorldObject::GetMapHeight(float x, float y, float z, bool vmap/* = true*/,
     return GetMap()->GetHeight(GetPhaseMask(), x, y, z, vmap, distanceToSearch, GetCollisionHeight());
 }
 
+void WorldObject::SetTransport(Transport* transport)
+{ 
+    if (transport) 
+    {
+        m_movementInfo.transport.guid = transport->GetGUID();
+        // could it be possible to initialize the rest of the transport data (seat, time, offset) here?
+        AddUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+    else // if(!m_vehicle) ?
+    {
+        m_movementInfo.transport.Reset();
+        RemoveUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+
+    m_transport = transport;
+}
+
+MovementInfo WorldObject::GetMovementInfo() const
+{
+    MovementInfo mInfo;
+    mInfo.guid = GetGUID();
+    mInfo.SetMovementFlags(GetUnitMovementFlags());
+    mInfo.SetExtraMovementFlags(GetExtraUnitMovementFlags());
+    mInfo.time = m_movementInfo.time;
+    mInfo.pos.Relocate
```

Comment:
codestyle

Reply:
Took me a moment to notice what you mean.
Needs a separating blank line above.